src/dcqc/suites/suite_abc.py

```diff
@@ -0,0 +1,157 @@
+        required_test_names = [test.__name__ for test in required_tests]
+        return required_test_names
+
+    def init_test_classes(self):
+        test_classes = self.list_test_classes()
+        tests = []
+        for test_cls in test_classes:
+            test_name = test_cls.__name__
+            skip = test_name in self.skipped_tests
+            test = test_cls(self.target, skip)
+            tests.append(test)
+        return tests
+
+    @classmethod
+    def _list_subclasses(cls) -> set[Type[SuiteABC]]:
+        subclasses = set(cls.__subclasses__())
+        subsubclasses_list = [subcls._list_subclasses() for subcls in subclasses]
+        subsubclasses = chain(*subsubclasses_list)
+        return subclasses.union(subsubclasses)
+
+    @classmethod
+    def get_suite_by_name(cls, name: str) -> Type[SuiteABC]:
+        subclasses = cls._list_subclasses()
+        registry = {subcls.__name__: subcls for subcls in subclasses}
+        if name not in registry:
+            options = list(registry)
+            message = f"Suite ({name}) not available ({options})."
+            raise ValueError(message)
+        return registry[name]
+
+    @overload
+    @classmethod
+    def get_suite_by_file_type(cls, file_type: str) -> Type[SuiteABC]:
+        """"""
+
+    @overload
+    @classmethod
+    def get_suite_by_file_type(cls, file_type: FileType) -> Type[SuiteABC]:
+        """"""
+
+    @classmethod
+    def get_suite_by_file_type(cls, file_type) -> Type[SuiteABC]:
+        if isinstance(file_type, str):
+            try:
+                file_type = FileType.get_file_type(file_type)
+            except ValueError:
+                file_type = FileType.get_file_type("*")
+        name = file_type.name
+        subclasses = cls._list_subclasses()
+        registry = {subcls.file_type.name: subcls for subcls in subclasses}
+        if name not in registry:
+            # TODO: This might have to be changed if we introduce
+            #       composite file types (e.g., BAM/BAI file pair)
+            return registry["*"]
+        return registry[name]
+
+    def compute_tests(self):
+        for test in self.tests:
+            test.get_status()
+
+    def compute_status(self) -> TestStatus:
+        self.compute_tests()
+        suite_status = TestStatus.NONE
+        for test in self.tests:
+            test_name = test.type
+            if test_name not in self.required_tests:
+                continue
+            test_status = test.get_status()
+            suite_status = test_status
+            if suite_status == TestStatus.FAIL:
+                break
+        return suite_status
+
+    def to_dict(self):
+        suite_status = self.compute_status()
+        test_dicts = []
+        for test in self.tests:
+            test_dict = test.to_dict()
+            test_dict.pop("target", None)  # Remove redundant `target` info
+            test_dicts.append(test_dict)
+        suite_dict = {
```

Comment:
Should this structure be captured in a python class in the future?

Reply:
We could follow the [Memento design pattern](https://refactoring.guru/design-patterns/memento), which uses a class to capture the instance state. In this case, I opted against it (for now) because these dictionaries are only mainly used for serialization. 

If they prove to be burdensome to maintain, we could look into creating simple `dataclasses` for passing this information along.

Reply:
Thats an interesting design pattern you bring up - I think this is great for now.

Were you thinking that you would use the memento design pattern to track the different validation statuses of a target?